lib/util.js

```diff
@@ -430,3 +432,149 @@ module.exports = exports = {
+    const nr = internalBinding('node_report');
+
+    // keep it un-exposed; lest programs play with it
+    // leaving us with a lot of unwanted sanity checks.
+    var config = {
+      events: [],
+      signal: 'SIGUSR2',
+      filename: '',
+      path: '',
+      verbose: false
+    };
+    const report = {
+      setDiagnosticReportOptions(options) {
+        // reuse the null and undefined checks. Save
+        // space when dealing with large number of arguments.
+        const list = parseOptions(options);
+
+        // flush the stale entries from report, as
+        // we are refreshing it, items that the users did not
+        // touch may be hanging around stale otherwise.
+        config = {};
+
+        // the parseOption method returns an array that include
+        // the indices at which valid params are present.
+        list.forEach((i) => {
+          switch (i) {
+            case '1':
+              if (Array.isArray(options.events))
+                config.events = options.events;
+              else
+                throw new ERR_INVALID_ARG_TYPE('events',
+                                               'Array',
+                                               options.events);
+              break;
+            case '2':
+              if (typeof options.signal !== 'string')
+                throw new ERR_INVALID_ARG_TYPE('signal',
+                                               'String',
+                                               options.signal);
+              process.removeListener(config.signal, handleSignal);
+              if (config.events.includes('signal'))
+                process.on(options.signal, handleSignal);
+              config.signal = options.signal;
+              break;
+            case '3':
+              if (typeof options.filename !== 'string')
+                throw new ERR_INVALID_ARG_TYPE('filename',
+                                               'String',
+                                               options.filename);
+              config.filename = options.filename;
+              break;
+            case '4':
+              if (typeof options.path !== 'string')
+                throw new ERR_INVALID_ARG_TYPE('path', 'String', options.path);
+              config.path = options.path;
+              break;
+            case '5':
+              if (typeof options.verbose !== 'string' &&
+                  typeof options.verbose !== 'boolean')
+                throw new ERR_INVALID_ARG_TYPE('verbose',
+                                               'Booelan | String' +
+                                               ' (true|false|yes|no)',
+                                               options.verbose);
+              config.verbose = options.verbose;
+              break;
+          }
+        });
+        // upload this new config to C++ land
+        nr.syncConfig(config, true);
+      },
+
+
+      triggerReport: function(file, err) {
```

Comment:
The other functions in this export can use the shorthand syntax also... e.g.

```js
triggerReport(file, err) { ...
```

Reply:
@jasnell - done, thanks & sorry for missing this.